broker.go

```diff
@@ -1570,7 +1570,7 @@ func (b *Broker) createSaslAuthenticateRequest(msg []byte) *SaslAuthenticateRequ
 func buildClientFirstMessage(token *AccessToken) ([]byte, error) {
 	var ext string
 
-	if token.Extensions != nil && len(token.Extensions) > 0 {
+	if token != nil && token.Extensions != nil && len(token.Extensions) > 0 {
```

Comment:
The `token.Extensions != nil` test is unnecessary as well. Since `token.Extensions == nil` implies `len(token.Extensions) == 0`.